Added RNTupleTempSource and RNTupleTempOutputModule

Dr15Jones opened the pull request:
#### PR description:

Created RNTuple based Source and OutputModules which are just the code from the Pool equivalents copied and then TTree replaced with RNTuple. This is meant to function as an intermediate solution for the TTree to RNTuple migration while we work on the final version.

#### PR validation:

All related unit tests pass.

resolves https://github.com/cms-sw/framework-team/issues/1555
resolves https://github.com/cms-sw/framework-team/issues/1557
resolves https://github.com/cms-sw/framework-team/issues/1556
resolves https://github.com/cms-sw/framework-team/issues/1558

Dr15Jones committed:
All RNTupleTemp related files

Names have changed but still using TTree for now.

Dr15Jones committed:
Replaced meta data TTrees with RNTuple

- Also removed some backwards compatibility code

Dr15Jones committed:
Added FWIO/RNTupleTempTests items
  - based on IOPool/Tests
  - includes changes needed to make tests work

Dr15Jones committed:
Removed cloning code from RNTuple implementation

Cloning is not an option for RNTuple.

Dr15Jones committed:
Moved output to use only RNTuple

Dr15Jones committed:
Simplifying input code for RNTuple migration

Dr15Jones committed:
edmRNTupleTempFileUtil now complete

Dr15Jones committed:
Updated edmRNTupleTempProvDump to only use RNTuple

Removed all references to TTree used during the transition.

Dr15Jones committed:
Removed unneeded header file

Dr15Jones committed:
Move input to only use RNTuple

cmsbuild commented:
cms-bot internal usage<!-- bot cache: {"commits":{"349907037b4cc63015638570c0207470eaca2897":{"files":["FWIO/RNTupleTempInput/bin/CollUtil.cc","FWIO/RNTupleTempInput/bin/EdmRNTupleTempFileUtil.cpp"],"squashed":false,"time":1759345409},"4e8c72d4222408195bfd7dbd827c450633f38cac":{"files":["FWIO/RNTupleTempInput/BuildFile.xml","FWIO/RNTupleTempInput/src/DuplicateChecker.cc","FWIO/RNTupleTempInput/src/DuplicateChecker.h","FWIO/RNTupleTempInput/src/EmbeddedRNTupleTempSource.cc","FWIO/RNTupleTempInput/src/EmbeddedRNTupleTempSource.h","FWIO/RNTupleTempInput/src/FixMissingStreamerInfos.cc","FWIO/RNTupleTempInput/src/InputFile.cc","FWIO/RNTupleTempInput/src/InputFile.h","FWIO/RNTupleTempInput/src/Module.cc","FWIO/RNTupleTempInput/src/ProvenanceAdaptor.cc","FWIO/RNTupleTempInput/src/ProvenanceAdaptor.h","FWIO/RNTupleTempInput/src/RNTupleTempSource.cc","FWIO/RNTupleTempInput/src/RNTupleTempSource.h","FWIO/RNTupleTempInput/src/RootDelayedReader.cc","FWIO/RNTupleTempInput/src/RootDelayedReader.h","FWIO/RNTupleTempInput/src/RootDelayedReaderBase.h","FWIO/RNTupleTempInput/src/RootEmbeddedFileSequence.cc","FWIO/RNTupleTempInput/src/RootEmbeddedFileSequence.h","FWIO/RNTupleTempInput/src/RootFile.cc","FWIO/RNTupleTempInput/src/RootFile.h","FWIO/RNTupleTempInput/src/RootInputFileSequence.cc","FWIO/RNTupleTempInput/src/RootInputFileSequence.h","FWIO/RNTupleTempInput/src/RootPrimaryFileSequence.cc","FWIO/RNTupleTempInput/src/RootPrimaryFileSequence.h","FWIO/RNTupleTempInput/src/RootPromptReadDelayedReader.cc","FWIO/RNTupleTempInput/src/RootPromptReadDelayedReader.h","FWIO/RNTupleTempInput/src/RootRNTuple.cc","FWIO/RNTupleTempInput/src/RootRNTuple.h","FWIO/RNTupleTempInput/src/RootSecondaryFileSequence.cc","FWIO/RNTupleTempInput/src/RootSecondaryFileSequence.h","FWIO/RNTupleTempInput/test/BuildFile.xml","FWIO/RNTupleTempInput/test/HeteroMerge_cfg.py","FWIO/RNTupleTempInput/test/Pool2FileInputTest_cfg.py","FWIO/RNTupleTempInput/test/PoolAliasTestStep1C_cfg.py","FWIO/RNTupleTempInput/test/PoolAliasTestStep1_DifferentOrder_cfg.py","FWIO/RNTupleTempInput/test/PoolAliasTestStep1_cfg.py","FWIO/RNTupleTempInput/test/PoolAliasTestStep2A_cfg.py","FWIO/RNTupleTempInput/test/PoolAliasTestStep2C_cfg.py","FWIO/RNTupleTempInput/test/PoolAliasTestStep2_DifferentOrder_cfg.py","FWIO/RNTupleTempInput/test/PoolAliasTestStep2_cfg.py","FWIO/RNTupleTempInput/test/PoolEmptyTest2_cfg.py","FWIO/RNTupleTempInput/test/PoolEmptyTest_cfg.py","FWIO/RNTupleTempInput/test/PoolGUIDTest_cfg.py","FWIO/RNTupleTempInput/test/PoolInputTest2_cfg.py","FWIO/RNTupleTempInput/test/PoolInputTest3_cfg.py","FWIO/RNTupleTempInput/test/PoolInputTest_cfg.py","FWIO/RNTupleTempInput/test/PoolInputTest_noDelay_cfg.py","FWIO/RNTupleTempInput/test/PoolInputTest_skipBadFiles_cfg.py","FWIO/RNTupleTempInput/test/PoolInputTest_skip_with_failure_cfg.py","FWIO/RNTupleTempInput/test/PoolNoParentDictionaryTestStep1_cfg.py","FWIO/RNTupleTempInput/test/PoolNoParentDictionaryTestStep2_cfg.py","FWIO/RNTupleTempInput/test/PrePool2FileInputTest_cfg.py","FWIO/RNTupleTempInput/test/PrePoolInputTest2_cfg.py","FWIO/RNTupleTempInput/test/PrePoolInputTest_cfg.py","FWIO/RNTupleTempInput/test/RunPerLumiTest_cfg.py","FWIO/RNTupleTempInput/test/TestPoolInput.sh","FWIO/RNTupleTempInput/test/TestPoolInputRunPerLumi.sh","FWIO/RNTupleTempInput/test/firstLuminosityBlockForEachRunTest_cfg.py","FWIO/RNTupleTempInput/test/firstLuminosityBlockForEachRun_skipLumis2_Test_cfg.py","FWIO/RNTupleTempInput/test/firstLuminosityBlockForEachRun_skipLumis_Test_cfg.py","FWIO/RNTupleTempInput/test/preMerge2_cfg.py","FWIO/RNTupleTempInput/test/preMerge_cfg.py","FWIO/RNTupleTempInput/test/sitelocalconfig/noFallbackFile/local/storage.json","FWIO/RNTupleTempInput/test/sitelocalconfig/noFallbackFile/site-local-config.xml","FWIO/RNTupleTempInput/test/sitelocalconfig/useFallbackFile/local/storage.json","FWIO/RNTupleTempInput/test/sitelocalconfig/useFallbackFile/site-local-config.xml","FWIO/RNTupleTempInput/test/testFileOpenErrorExitCode.sh","FWIO/RNTupleTempInput/test/testForStreamerInfo.C","FWIO/RNTupleTempInput/test/testNoParentDictionary.sh","FWIO/RNTupleTempInput/test/testReducedProcessHistoryCreate_cfg.py","FWIO/RNTupleTempInput/test/testReducedProcessHistoryHardwareResources.sh","FWIO/RNTupleTempInput/test/testReducedProcessHistoryVersion.sh","FWIO/RNTupleTempInput/test/testReducedProcessHistory_cfg.py","FWIO/RNTupleTempInput/test/testRefProductIDMetadataConsistencyRoot.sh","FWIO/RNTupleTempInput/test/testRefProductIDMetadataConsistencyRootMerge_cfg.py","FWIO/RNTupleTempInput/test/testRefProductIDMetadataConsistencyRootTest_cfg.py","FWIO/RNTupleTempInput/test/testRefProductIDMetadataConsistencyRoot_cfg.py","FWIO/RNTupleTempInput/test/test_fileOpenErrorExitCode_cfg.py","FWIO/RNTupleTempInput/test/test_make_multi_lumi_cfg.py","FWIO/RNTupleTempInput/test/test_make_overlapping_lumis_cfg.py","FWIO/RNTupleTempInput/test/test_merge_two_files.py","FWIO/RNTupleTempInput/test/test_read_overlapping_lumis_cfg.py","FWIO/RNTupleTempInput/test/unit_test_outputs/RunPerLumiTest.filtered.txt","FWIO/RNTupleTempOutput/BuildFile.xml","FWIO/RNTupleTempOutput/interface/RNTupleTempOutputModule.h","FWIO/RNTupleTempOutput/interface/TimeoutRNTupleTempOutputModule.h","FWIO/RNTupleTempOutput/plugins/BuildFile.xml","FWIO/RNTupleTempOutput/plugins/Module.cc","FWIO/RNTupleTempOutput/src/RNTupleTempOutputModule.cc","FWIO/RNTupleTempOutput/src/RootOutputFile.cc","FWIO/RNTupleTempOutput/src/RootOutputFile.h","FWIO/RNTupleTempOutput/src/RootOutputRNTuple.cc","FWIO/RNTupleTempOutput/src/RootOutputRNTuple.h","FWIO/RNTupleTempOutput/src/TimeoutRNTupleTempOutputModule.cc","FWIO/RNTupleTempOutput/test/BuildFile.xml","FWIO/RNTupleTempOutput/test/PoolDropRead_cfg.py","FWIO/RNTupleTempOutput/test/PoolDropTest_cfg.py","FWIO/RNTupleTempOutput/test/PoolMissingRead_cfg.py","FWIO/RNTupleTempOutput/test/PoolMissingTest_cfg.py","FWIO/RNTupleTempOutput/test/PoolOutputAliasTestCheckResults.py","FWIO/RNTupleTempOutput/test/PoolOutputAliasTest_cfg.py","FWIO/RNTupleTempOutput/test/PoolOutputEmptyEventsTest_cfg.py","FWIO/RNTupleTempOutput/test/PoolOutputMergeWithEmptyFile_cfg.py","FWIO/RNTupleTempOutput/test/PoolOutputRead_cfg.py","FWIO/RNTupleTempOutput/test/PoolOutputTestOverrideGUID_cfg.py","FWIO/RNTupleTempOutput/test/PoolOutputTestUnscheduledRead_cfg.py","FWIO/RNTupleTempOutput/test/PoolOutputTestUnscheduled_cfg.py","FWIO/RNTupleTempOutput/test/PoolOutputTest_cfg.py","FWIO/RNTupleTempOutput/test/PoolTransientRead_cfg.py","FWIO/RNTupleTempOutput/test/PoolTransientTest_cfg.py","FWIO/RNTupleTempOutput/test/RNTupleTempOutputAliasTest.sh","FWIO/RNTupleTempOutput/test/TestProvA_cfg.py","FWIO/RNTupleTempOutput/test/TestProvB_cfg.py","FWIO/RNTupleTempOutput/test/TestProvC_cfg.py","FWIO/RNTupleTempOutput/test/TestRNTupleTempOutput.sh"],"squashed":false,"time":1758714732},"7127ae81e3d712a6ee59729b7920f8e5581b5684":{"files":["FWIO/RNTupleTempInput/src/InputFile.h","FWIO/RNTupleTempInput/src/RootDelayedReader.cc","FWIO/RNTupleTempInput/src/RootDelayedReader.h","FWIO/RNTupleTempInput/src/RootEmbeddedFileSequence.cc","FWIO/RNTupleTempInput/src/RootFile.cc","FWIO/RNTupleTempInput/src/RootPrimaryFileSequence.cc","FWIO/RNTupleTempInput/src/RootPromptReadDelayedReader.cc","FWIO/RNTupleTempInput/src/RootPromptReadDelayedReader.h","FWIO/RNTupleTempInput/src/RootRNTuple.cc","FWIO/RNTupleTempInput/src/RootRNTuple.h","FWIO/RNTupleTempOutput/src/RootOutputRNTuple.cc","FWIO/RNTupleTempOutput/src/RootOutputRNTuple.h"],"squashed":false,"time":1759345320},"9c5d03b083481c74ffe4f10c3eeb96a8f92e3642":{"files":["FWIO/RNTupleTempOutput/interface/RNTupleTempOutputModule.h","FWIO/RNTupleTempOutput/src/RNTupleTempOutputModule.cc","FWIO/RNTupleTempOutput/src/RootOutputFile.cc","FWIO/RNTupleTempOutput/src/RootOutputFile.h","FWIO/RNTupleTempOutput/src/RootOutputRNTuple.cc","FWIO/RNTupleTempOutput/src/RootOutputRNTuple.h"],"squashed":false,"time":1759259746},"c060e8c34dec7b35acafcb455eb158cfdb67cb4f":{"files":["FWIO/RNTuple/plugins/RNTupleOutputModule.cc","FWIO/RNTupleTempInput/src/RNTupleTempSource.cc","FWIO/RNTupleTempInput/src/RootFile.cc","FWIO/RNTupleTempInput/src/RootFile.h","FWIO/RNTupleTempOutput/interface/RNTupleTempOutputModule.h","FWIO/RNTupleTempOutput/src/RNTupleTempOutputModule.cc","FWIO/RNTupleTempOutput/src/RootOutputFile.cc","FWIO/RNTupleTempOutput/src/RootOutputFile.h","FWIO/RNTupleTempOutput/src/RootOutputRNTuple.cc","FWIO/RNTupleTempOutput/src/RootOutputRNTuple.h","FWIO/RNTupleTempTests/test/run_RunMerge.sh"],"squashed":false,"time":1759154092},"c4e7725902d2b7c1361f5346052068dda0ce8f42":{"files":["FWIO/RNTupleTempOutput/src/RNTupleTempOutputModule.cc"],"squashed":false,"time":1759576092},"df7cf2e235ffb1230f1187c79531ba309c839c40":{"files":["FWIO/RNTupleTempInput/bin/BuildFile.xml","FWIO/RNTupleTempInput/bin/CollUtil.cc","FWIO/RNTupleTempInput/bin/CollUtil.h","FWIO/RNTupleTempInput/bin/EdmRNTupleTempFileUtil.cpp","FWIO/RNTupleTempInput/bin/EdmRNTupleTempProvDump.cc","FWIO/RNTupleTempInput/src/InputFile.h","FWIO/RNTupleTempInput/src/RootFile.cc","FWIO/RNTupleTempInput/src/RootFile.h","FWIO/RNTupleTempInput/test/testReducedProcessHistoryHardwareResources.sh","FWIO/RNTupleTempInput/test/testReducedProcessHistoryVersion.sh","FWIO/RNTupleTempOutput/interface/RNTupleTempOutputModule.h","FWIO/RNTupleTempOutput/src/RNTupleTempOutputModule.cc","FWIO/RNTupleTempOutput/src/RootOutputFile.cc","FWIO/RNTupleTempOutput/src/RootOutputFile.h","FWIO/RNTupleTempOutput/test/TestRNTupleTempOutput.sh"],"squashed":false,"time":1758714733},"e190c7a44708508719b555da131ed0ef5f960b4e":{"files":["FWIO/RNTupleTempInput/src/RootDelayedReader.cc","FWIO/RNTupleTempInput/src/RootDelayedReader.h","FWIO/RNTupleTempInput/src/RootFile.cc","FWIO/RNTupleTempInput/src/RootFile.h","FWIO/RNTupleTempInput/src/RootPromptReadDelayedReader.cc","FWIO/RNTupleTempInput/src/RootPromptReadDelayedReader.h","FWIO/RNTupleTempInput/src/RootRNTuple.cc","FWIO/RNTupleTempInput/src/RootRNTuple.h","FWIO/RNTupleTempOutput/test/BuildFile.xml","FWIO/RNTupleTempTests/test/run_TestProcessBlock.sh","FWIO/RNTupleTempTests/test/testLooperEventNavigation2_cfg.py","FWIO/RNTupleTempTests/test/testLooperEventNavigation3_cfg.py","FWIO/RNTupleTempTests/test/testProcessBlockMergeOfMergedFiles_cfg.py","FWIO/RNTupleTempTests/test/testProcessBlockNOMergeOfMergedFiles_cfg.py","IOPool/Input/src/RootFile.cc"],"squashed":false,"time":1759576124},"ee776c9d31010a32799b5fbe43dd22e0b1b432b1":{"files":["FWIO/RNTupleTempInput/bin/CollUtil.cc","FWIO/RNTupleTempInput/bin/CollUtil.h","FWIO/RNTupleTempInput/bin/EdmRNTupleTempFileUtil.cpp","FWIO/RNTupleTempInput/src/RootFile.cc","FWIO/RNTupleTempInput/src/RootFile.h","FWIO/RNTupleTempOutput/src/RootOutputFile.cc","FWIO/RNTupleTempTests/test/AssociationMapReadTest_cfg.py","FWIO/RNTupleTempTests/test/AssociationMapTest_cfg.py","FWIO/RNTupleTempTests/test/BuildFile.xml","FWIO/RNTupleTempTests/test/DetSetVectorThinningTest1_cfg.py","FWIO/RNTupleTempTests/test/DetSetVectorThinningTest2_cfg.py","FWIO/RNTupleTempTests/test/EventHistory_1_cfg.py","FWIO/RNTupleTempTests/test/EventHistory_2_cfg.py","FWIO/RNTupleTempTests/test/EventHistory_3_cfg.py","FWIO/RNTupleTempTests/test/EventHistory_4_cfg.py","FWIO/RNTupleTempTests/test/EventHistory_5_cfg.py","FWIO/RNTupleTempTests/test/EventHistory_6_cfg.py","FWIO/RNTupleTempTests/test/Run_MissingDictionary.sh","FWIO/RNTupleTempTests/test/SlimmingTest1_cfg.py","FWIO/RNTupleTempTests/test/SlimmingTest2A_cfg.py","FWIO/RNTupleTempTests/test/SlimmingTest2B_cfg.py","FWIO/RNTupleTempTests/test/SlimmingTest2C_cfg.py","FWIO/RNTupleTempTests/test/SlimmingTest2D_cfg.py","FWIO/RNTupleTempTests/test/SlimmingTest2E_cfg.py","FWIO/RNTupleTempTests/test/SlimmingTest2F_cfg.py","FWIO/RNTupleTempTests/test/SlimmingTest2G_cfg.py","FWIO/RNTupleTempTests/test/SlimmingTest2H_cfg.py","FWIO/RNTupleTempTests/test/SlimmingTest2I_cfg.py","FWIO/RNTupleTempTests/test/SlimmingTest3B_cfg.py","FWIO/RNTupleTempTests/test/SlimmingTest3C_cfg.py","FWIO/RNTupleTempTests/test/SlimmingTest3D_cfg.py","FWIO/RNTupleTempTests/test/SlimmingTest3E_cfg.py","FWIO/RNTupleTempTests/test/SlimmingTest3F_cfg.py","FWIO/RNTupleTempTests/test/SlimmingTest3I_cfg.py","FWIO/RNTupleTempTests/test/SlimmingTest4B_cfg.py","FWIO/RNTupleTempTests/test/SlimmingTest4F_cfg.py","FWIO/RNTupleTempTests/test/ThinningTest1_cfg.py","FWIO/RNTupleTempTests/test/ThinningTest2_cfg.py","FWIO/RNTupleTempTests/test/ThinningTest3_cfg.py","FWIO/RNTupleTempTests/test/ThinningTest4Slimming_cfg.py","FWIO/RNTupleTempTests/test/ThinningTest4_cfg.py","FWIO/RNTupleTempTests/test/ThinningTest_dropOnInput_cfg.py","FWIO/RNTupleTempTests/test/concurrentIOVsAndRuns.sh","FWIO/RNTupleTempTests/test/eventHistoryTest.sh","FWIO/RNTupleTempTests/test/inconsistent_products_prod_cfg.py","FWIO/RNTupleTempTests/test/inconsistent_products_test_cfg.py","FWIO/RNTupleTempTests/test/makeEmptyRootFile.py","FWIO/RNTupleTempTests/test/provenance_check_cfg.py","FWIO/RNTupleTempTests/test/provenance_prod_cfg.py","FWIO/RNTupleTempTests/test/provenance_test.sh","FWIO/RNTupleTempTests/test/ref_alias_compare_drop_alias_cfg.py","FWIO/RNTupleTempTests/test/ref_alias_compare_drop_original_cfg.py","FWIO/RNTupleTempTests/test/ref_alias_compare_read_alias_cfg.py","FWIO/RNTupleTempTests/test/ref_alias_compare_read_original_cfg.py","FWIO/RNTupleTempTests/test/ref_merge_cfg.py","FWIO/RNTupleTempTests/test/ref_merge_prod_cfg.py","FWIO/RNTupleTempTests/test/ref_merge_test_cfg.py","FWIO/RNTupleTempTests/test/run_RefAlias.sh","FWIO/RNTupleTempTests/test/run_RefMerge.sh","FWIO/RNTupleTempTests/test/run_RunMerge.sh","FWIO/RNTupleTempTests/test/run_RunMerge2.sh","FWIO/RNTupleTempTests/test/run_SeriesOfProcesses.sh","FWIO/RNTupleTempTests/test/run_TestDropOnInput.sh","FWIO/RNTupleTempTests/test/run_TestEDAlias.sh","FWIO/RNTupleTempTests/test/run_TestExistingDictionary.sh","FWIO/RNTupleTempTests/test/run_TestFallback.sh","FWIO/RNTupleTempTests/test/run_TestGetBy.sh","FWIO/RNTupleTempTests/test/run_TestGetByLabel.sh","FWIO/RNTupleTempTests/test/run_TestOutput.sh","FWIO/RNTupleTempTests/test/run_TestProcessBlock.sh","FWIO/RNTupleTempTests/test/run_ThinningTests.sh","FWIO/RNTupleTempTests/test/run_inconsistent_products.sh","FWIO/RNTupleTempTests/test/run_testMaxEventsOutput.sh","FWIO/RNTupleTempTests/test/run_unscheduledFailOnOutput.sh","FWIO/RNTupleTempTests/test/testConcurrentIOVsAndRunsRead_cfg.py","FWIO/RNTupleTempTests/test/testConcurrentIOVsAndRuns_cfg.py","FWIO/RNTupleTempTests/test/testDropOnInput1_1_cfg.py","FWIO/RNTupleTempTests/test/testDropOnInput1_2_cfg.py","FWIO/RNTupleTempTests/test/testDropOnInput2_cfg.py","FWIO/RNTupleTempTests/test/testDropOnInput3_cfg.py","FWIO/RNTupleTempTests/test/testDropOnInputRead2001_cfg.py","FWIO/RNTupleTempTests/test/testDropOnInputRead2_cfg.py","FWIO/RNTupleTempTests/test/testDropOnInputRead3_cfg.py","FWIO/RNTupleTempTests/test/testDuplicateProcess_cfg.py","FWIO/RNTupleTempTests/test/testEDAliasAnalyze_cfg.py","FWIO/RNTupleTempTests/test/testEDAliasTask_cfg.py","FWIO/RNTupleTempTests/test/testExistingDictionaryCheckingRead_cfg.py","FWIO/RNTupleTempTests/test/testExistingDictionaryChecking_cfg.py","FWIO/RNTupleTempTests/test/testGetBy1Mod_cfg.py","FWIO/RNTupleTempTests/test/testGetBy1_cfg.py","FWIO/RNTupleTempTests/test/testGetBy2_cfg.py","FWIO/RNTupleTempTests/test/testGetBy3_cfg.py","FWIO/RNTupleTempTests/test/testGetByLabelStep1_cfg.py","FWIO/RNTupleTempTests/test/testGetByLabelStep2_cfg.py","FWIO/RNTupleTempTests/test/testGetByMerge_cfg.py","FWIO/RNTupleTempTests/test/testGetByRunsLumisMode_cfg.py","FWIO/RNTupleTempTests/test/testGetByRunsMode_cfg.py","FWIO/RNTupleTempTests/test/testGetByWithEmptyRun_cfg.py","FWIO/RNTupleTempTests/test/testLooperEventNavigation.txt","FWIO/RNTupleTempTests/test/testLooperEventNavigation1_cfg.py","FWIO/RNTupleTempTests/test/testLooperEventNavigation2.txt","FWIO/RNTupleTempTests/test/testLooperEventNavigation2_cfg.py","FWIO/RNTupleTempTests/test/testLooperEventNavigation3.txt","FWIO/RNTupleTempTests/test/testLooperEventNavigation3_cfg.py","FWIO/RNTupleTempTests/test/testLooperEventNavigation_cfg.py","FWIO/RNTupleTempTests/test/testMaxEventsOutput_cfg.py","FWIO/RNTupleTempTests/test/testMaybeUninitializedIntProductPart1_cfg.py","FWIO/RNTupleTempTests/test/testMaybeUninitializedIntProductPart2_cfg.py","FWIO/RNTupleTempTests/test/testMissingDictionaryChecking_cfg.py","FWIO/RNTupleTempTests/test/testNoProcessFallback1_cfg.py","FWIO/RNTupleTempTests/test/testNoProcessFallback2_cfg.py","FWIO/RNTupleTempTests/test/testNoProcessFallback3_cfg.py","FWIO/RNTupleTempTests/test/testNoProcessFallbackNoCurrent_cfg.py","FWIO/RNTupleTempTests/test/testOutput1_cfg.py","FWIO/RNTupleTempTests/test/testOutput2_cfg.py","FWIO/RNTupleTempTests/test/testProcessBlock1_cfg.py","FWIO/RNTupleTempTests/test/testProcessBlock2Dropped_cfg.py","FWIO/RNTupleTempTests/test/testProcessBlock2_cfg.py","FWIO/RNTupleTempTests/test/testProcessBlock3_cfg.py","FWIO/RNTupleTempTests/test/testProcessBlock4_cfg.py","FWIO/RNTupleTempTests/test/testProcessBlock5_cfg.py","FWIO/RNTupleTempTests/test/testProcessBlockDropOnInput_cfg.py","FWIO/RNTupleTempTests/test/testProcessBlockDropOnOutput2_cfg.py","FWIO/RNTupleTempTests/test/testProcessBlockDropOnOutput_cfg.py","FWIO/RNTupleTempTests/test/testProcessBlockDummy_cfg.py","FWIO/RNTupleTempTests/test/testProcessBlockFailMerge_cfg.py","FWIO/RNTupleTempTests/test/testProcessBlockLooper_cfg.py","FWIO/RNTupleTempTests/test/testProcessBlockMerge2_cfg.py","FWIO/RNTupleTempTests/test/testProcessBlockMerge3_cfg.py","FWIO/RNTupleTempTests/test/testProcessBlockMergeOfMergedFiles2_cfg.py","FWIO/RNTupleTempTests/test/testProcessBlockMergeOfMergedFiles_cfg.py","FWIO/RNTupleTempTests/test/testProcessBlockMerge_cfg.py","FWIO/RNTupleTempTests/test/testProcessBlockNOMergeOfMergedFiles_cfg.py","FWIO/RNTupleTempTests/test/testProcessBlockNonStrict2_cfg.py","FWIO/RNTupleTempTests/test/testProcessBlockNonStrict3_cfg.py","FWIO/RNTupleTempTests/test/testProcessBlockNonStrict_cfg.py","FWIO/RNTupleTempTests/test/testProcessBlockRead2_cfg.py","FWIO/RNTupleTempTests/test/testProcessBlockReadDropOnOutput2_cfg.py","FWIO/RNTupleTempTests/test/testProcessBlockReadDropOnOutput_cfg.py","FWIO/RNTupleTempTests/test/testProcessBlockReadThreeFileInput_cfg.py","FWIO/RNTupleTempTests/test/testProcessBlockRead_cfg.py","FWIO/RNTupleTempTests/test/testProcessBlockTEST_cfg.py","FWIO/RNTupleTempTests/test/testProcessBlockThreeFileInput_cfg.py","FWIO/RNTupleTempTests/test/testRunMergeCOPY100_cfg.py","FWIO/RNTupleTempTests/test/testRunMergeCOPY101_cfg.py","FWIO/RNTupleTempTests/test/testRunMergeCOPY1_cfg.py","FWIO/RNTupleTempTests/test/testRunMergeCOPY_cfg.py","FWIO/RNTupleTempTests/test/testRunMergeFastCloning_cfg.py","FWIO/RNTupleTempTests/test/testRunMergeMERGE100_cfg.py","FWIO/RNTupleTempTests/test/testRunMergeMERGE101_cfg.py","FWIO/RNTupleTempTests/test/testRunMergeMERGE102_cfg.py","FWIO/RNTupleTempTests/test/testRunMergeMERGE1_cfg.py","FWIO/RNTupleTempTests/test/testRunMergeMERGE2_cfg.py","FWIO/RNTupleTempTests/test/testRunMergeMERGE3_cfg.py","FWIO/RNTupleTempTests/test/testRunMergeMERGE3x_cfg.py","FWIO/RNTupleTempTests/test/testRunMergeMERGE4_cfg.py","FWIO/RNTupleTempTests/test/testRunMergeMERGE5_cfg.py","FWIO/RNTupleTempTests/test/testRunMergeMERGE6_cfg.py","FWIO/RNTupleTempTests/test/testRunMergeMERGE_cfg.py","FWIO/RNTupleTempTests/test/testRunMergeNoRunLumiSort_cfg.py","FWIO/RNTupleTempTests/test/testRunMergePROD0_cfg.py","FWIO/RNTupleTempTests/test/testRunMergePROD100_cfg.py","FWIO/RNTupleTempTests/test/testRunMergePROD101_cfg.py","FWIO/RNTupleTempTests/test/testRunMergePROD102_cfg.py","FWIO/RNTupleTempTests/test/testRunMergePROD11_cfg.py","FWIO/RNTupleTempTests/test/testRunMergePROD1_cfg.py","FWIO/RNTupleTempTests/test/testRunMergePROD2EXTRA_cfg.py","FWIO/RNTupleTempTests/test/testRunMergePROD2_cfg.py","FWIO/RNTupleTempTests/test/testRunMergePROD3EXTRA_cfg.py","FWIO/RNTupleTempTests/test/testRunMergePROD3_cfg.py","FWIO/RNTupleTempTests/test/testRunMergePROD4_cfg.py","FWIO/RNTupleTempTests/test/testRunMergePROD5_cfg.py","FWIO/RNTupleTempTests/test/testRunMergePROD6_cfg.py","FWIO/RNTupleTempTests/test/testRunMergePROD7_cfg.py","FWIO/RNTupleTempTests/test/testRunMergePickEvents_cfg.py","FWIO/RNTupleTempTests/test/testRunMergePickEventsx_cfg.py","FWIO/RNTupleTempTests/test/testRunMergeSPLIT100_cfg.py","FWIO/RNTupleTempTests/test/testRunMergeSPLIT101_cfg.py","FWIO/RNTupleTempTests/test/testRunMergeSPLIT102_cfg.py","FWIO/RNTupleTempTests/test/testRunMergeSPLIT103_cfg.py","FWIO/RNTupleTempTests/test/testRunMergeTEST100_cfg.py","FWIO/RNTupleTempTests/test/testRunMergeTEST101_cfg.py","FWIO/RNTupleTempTests/test/testRunMergeTEST11_cfg.py","FWIO/RNTupleTempTests/test/testRunMergeTEST1_cfg.py","FWIO/RNTupleTempTests/test/testRunMergeTEST2_cfg.py","FWIO/RNTupleTempTests/test/testRunMergeTEST3_cfg.py","FWIO/RNTupleTempTests/test/testRunMergeTEST4_cfg.py","FWIO/RNTupleTempTests/test/testRunMergeTEST5_cfg.py","FWIO/RNTupleTempTests/test/testRunMergeTEST6_cfg.py","FWIO/RNTupleTempTests/test/testRunMergeTESTFAIL_cfg.py","FWIO/RNTupleTempTests/test/testRunMergeTEST_cfg.py","FWIO/RNTupleTempTests/test/testSeriesOfProcessesHLT_cfg.py","FWIO/RNTupleTempTests/test/testSeriesOfProcessesPROD2TEST_cfg.py","FWIO/RNTupleTempTests/test/testSeriesOfProcessesPROD2TEST_unscheduled_cfg.py","FWIO/RNTupleTempTests/test/testSeriesOfProcessesPROD3TEST_cfg.py","FWIO/RNTupleTempTests/test/testSeriesOfProcessesPROD_cfg.py","FWIO/RNTupleTempTests/test/testSeriesOfProcessesTEST1_cfg.py","FWIO/RNTupleTempTests/test/testSeriesOfProcessesTEST2_cfg.py","FWIO/RNTupleTempTests/test/testSeriesOfProcessesTEST3_cfg.py","FWIO/RNTupleTempTests/test/testSeriesOfProcessesTEST_cfg.py","FWIO/RNTupleTempTests/test/unit_test_outputs/testGetBy1.log","FWIO/RNTupleTempTests/test/unit_test_outputs/testGetBy2.log","FWIO/RNTupleTempTests/test/unit_test_outputs/testLooperEventNavigationOutput.txt","FWIO/RNTupleTempTests/test/unscheduled_fail_on_output_IgnoreCompletely_cfg.py","FWIO/RNTupleTempTests/test/unscheduled_fail_on_output_Rethrow_cfg.py","FWIO/RNTupleTempTests/test/unscheduled_fail_on_output_TryToContinue_cfg.py","FWIO/RNTupleTempTests/test/unscheduled_fail_on_output_cfg.py","FWIO/RNTupleTempTests/test/unscheduled_fail_on_output_no_dependency_TryToContinue_cfg.py","FWIO/RNTupleTempTests/test/unscheduled_fail_on_output_read_found_events.py","FWIO/RNTupleTempTests/test/unscheduled_fail_on_output_read_no_events.py","FWIO/RNTupleTempTests/test/useEmptyRootFile.py"],"squashed":false,"time":1758718913},"ffeefe8e57f912494035982ba049f5c0b9b10f72":{"files":["FWIO/RNTupleTempInput/bin/EdmRNTupleTempProvDump.cc"],"squashed":false,"time":1759406157}},"emoji":{},"last_seen_sha":"e190c7a44708508719b555da131ed0ef5f960b4e","signatures":{}} -->

cmsbuild commented:
A new Pull Request was created by  @Dr15Jones for CMSSW_16_0_RNTUPLE_X.

It involves the following packages:

- FWIO/RNTuple (**core**)
- FWIO/RNTupleTempInput (****)
- FWIO/RNTupleTempOutput (****)
- FWIO/RNTupleTempTests (****)
- IOPool/Input (**core**)


The following packages do not have a category, yet:

FWIO/RNTupleTempInput
FWIO/RNTupleTempOutput
FWIO/RNTupleTempTests
Please create a PR for https://github.com/cms-sw/cms-bot/blob/master/categories_map.py to assign category

@Dr15Jones, @makortel, @smuzaffar can you please review it and eventually sign? Thanks.
@makortel, @missirol, @mmusich, @wddgit this is something you requested to watch as well.
@ftenchini, @mandrenguyen, @sextonkennedy you are the release manager for this.

cms-bot commands are listed <a href="http://cms-sw.github.io/cms-bot-cmssw-cmds.html">here</a>

Dr15Jones commented:
please test

cmsbuild commented:
+code-checks

Logs: https://cmssdt.cern.ch/SDT/code-checks/cms-sw-PR-49059/46288

cmsbuild commented:
Pull request #49059 was updated. @Dr15Jones, @makortel, @smuzaffar can you please check and sign again.

Dr15Jones commented:
Please test

cmsbuild commented:
-1

**Failed Tests**: UnitTests 
**Summary**: https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-4e6978/48460/summary.html
**COMMIT**: c2014d4c316e536f9b4860e569dd173fee29adf5
**CMSSW**: CMSSW_16_0_RNTUPLE_X_2025-10-03-2300/el8_amd64_gcc13
**User test area**: For local testing, you can use `/cvmfs/cms-ci.cern.ch/week1/cms-sw/cmssw/49059/48460/install.sh` to create a dev area with all the needed externals and cmssw changes.


## Unit Tests

I found 1 errors in the following unit tests:

<pre>---> test TestFWIORNTupleTempInputNoParentDictionary had ERRORS
</pre>

## Comparison Summary



**Summary**:
 - You potentially [added 607 lines](https://cmssdt.cern.ch/SDT/jenkins-artifacts/baseLineComparisons/CMSSW_16_0_X_2025-10-03-2300+4e6978/70759/validateJR/logRootQA-events.log) to the logs
 - [Reco comparison results: 3 differences found in the comparisons](https://cmssdt.cern.ch/SDT/jenkins-artifacts/baseLineComparisons/CMSSW_16_0_X_2025-10-03-2300+4e6978/70759/validateJR.html)
 - DQMHistoTests: Total files compared: 51
 - DQMHistoTests: Total histograms compared: 3940073
 - DQMHistoTests: [Total failures: 429](https://cmssdt.cern.ch/SDT/jenkins-artifacts/baseLineComparisons/CMSSW_16_0_X_2025-10-03-2300+4e6978/70759/validateJR.html)
 - DQMHistoTests: Total nulls: 0
 - DQMHistoTests: Total successes: 3939624
 - DQMHistoTests: Total skipped: 20
 - DQMHistoTests: Total Missing objects: 0
 - DQMHistoSizes: Histogram memory added: 0.0 KiB( 50 files compared)
 - Checked 218 log files, 8 edm output root files, 51 DQM output files
 - [TriggerResults](https://cmssdt.cern.ch/SDT/jenkins-artifacts/baseLineComparisons/CMSSW_16_0_X_2025-10-03-2300+4e6978/70759/triggerResults): no differences found

cmsbuild commented:
Pull request #49059 was updated. @Dr15Jones, @cmsbuild, @makortel, @smuzaffar can you please check and sign again.

Dr15Jones commented:
please test

Dr15Jones commented:
new packages requested in https://github.com/cms-sw/cms-bot/pull/2586

Dr15Jones committed:
Better configuration of RNTupleTemp*

- made TTree only parameters optional
- added RNTuple specific options

cmsbuild commented:
+code-checks

Logs: https://cmssdt.cern.ch/SDT/code-checks/cms-sw-PR-49059/46308

- Found files with invalid states:
  - FWIO/RNTupleTempInput/test/testNoParentDictionary.sh:
    - Added: 4e8c72d4222408195bfd7dbd827c450633f38cac
    - Deleted: 9aa34afb2d9f130f493f3c8077645c063d8002bd

Dr15Jones committed:
Update to use std::format

Dr15Jones committed:
Handle RNTuple optimizations

- properly pass optimizations to RNTupleWriter
- can specify per Field optimizations

Dr15Jones commented:
please test

makortel commented on FWIO/RNTupleTempInput/test/TestPoolInput.sh:
Ideally this file and its components would be renamed `Pool` --> `RNTupleTemp`, but maybe that's not worth of the effort.

Dr15Jones replied:
I didn't think it was worth it.

makortel reviewed (commented):
Does the `RNTupleTempSource` support prompt reading in the same capacity as `PoolSource` via `delayReadingEventProducts=False`?

Dr15Jones commented on FWIO/RNTupleTempInput/src/RootFile.cc:
```diff
@@ -0,0 +1,2066 @@
+  }
+
+  void RootFile::updateFileBlock(FileBlock& fileBlock) {
+    std::vector<TTree*> processBlockTrees;
```
Again, it is part of the `FileBlock` API so have to declare this empty container.

cmsbuild commented:
This pull request is fully signed and it will be integrated in one of the next CMSSW_16_0_RNTUPLE_X IBs (tests are also fine) and once validation in the development release cycle CMSSW_16_0_X is complete. This pull request will now be reviewed by the release team before it's merged. @mandrenguyen, @ftenchini, @sextonkennedy (and backports should be raised in the release meeting by the corresponding L2)

Dr15Jones commented on FWIO/RNTupleTempOutput/src/RootOutputFile.cc:
```diff
@@ -0,0 +1,740 @@
+      model->AddField(std::move(field));
+    }
+    auto writeOptions = ROOT::RNTupleWriteOptions();
+    //writeOptions.SetCompression(convert(iConfig.compressionAlgo), iConfig.compressionLevel);
```
So this makes the ParameterSet RNTuple use the default settings for optimization, i.e. default compression and buffer sizes. We should probably pass through at least the compression settings from the configuration.

makortel replied:
Ok. Do I understand correctly then that with this PR the `RNTupleTempOutputModule` uses only the default write options, and all configuration options are effectively ignored?

> We should probably pass through at least the compression settings from the configuration.

How about following the logic in the prototype's `RNTupleOutputFile`? I.e.
https://github.com/cms-sw/cmssw/blob/8b9eeed99905a3b1b49d6df765c27b6a59bf548b/FWIO/RNTuple/plugins/RNTupleOutputFile.cc#L246-L255
for Events and
https://github.com/cms-sw/cmssw/blob/8b9eeed99905a3b1b49d6df765c27b6a59bf548b/FWIO/RNTuple/plugins/RNTupleOutputFile.cc#L200-L202
for the others?
